src/auto/input_stream.rs

```diff
@@ -0,0 +1,248 @@
+// This file was generated by gir (d8a605d) from gir-files (469db10)
+// DO NOT EDIT
+
+use Cancellable;
+use Error;
+use ffi;
+#[cfg(any(feature = "v2_34", feature = "dox"))]
+use glib;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct InputStream(Object<ffi::GInputStream, ffi::GInputStreamClass>);
+
+    match fn {
+        get_type => || ffi::g_input_stream_get_type(),
+    }
+}
+
+pub trait InputStreamExt {
+    fn clear_pending(&self);
+
+    fn close<'a, P: Into<Option<&'a Cancellable>>>(&self, cancellable: P) -> Result<(), Error>;
+
+    fn close_async<'a, P: Into<Option<&'a Cancellable>>, Q: Fn(Result<(), Error>) + Send + Sync + 'static>(&self, io_priority: i32, cancellable: P, callback: Q);
+
+    fn has_pending(&self) -> bool;
+
+    fn is_closed(&self) -> bool;
+
+    fn read<'a, P: Into<Option<&'a Cancellable>>>(&self, buffer: &[u8], cancellable: P) -> Result<isize, Error>;
+
+    fn read_all<'a, P: Into<Option<&'a Cancellable>>>(&self, buffer: &[u8], cancellable: P) -> Result<usize, Error>;
+
+    #[cfg(any(feature = "v2_44", feature = "dox"))]
+    fn read_all_async<'a, P: Into<Option<&'a Cancellable>>, Q: Fn(Result<usize, Error>) + Send + Sync + 'static>(&self, buffer: &[u8], io_priority: i32, cancellable: P, callback: Q);
+
+    fn read_async<'a, P: Into<Option<&'a Cancellable>>, Q: Fn(Result<(), Error>) + Send + Sync + 'static>(&self, buffer: &[u8], io_priority: i32, cancellable: P, callback: Q);
```

Comment:
And this buffer must even stay alive until the async callback is called

Reply:
There way to do it?

Reply:
With autogenerated code? No, not right now

Reply:
In manual code?

Reply:
Yes